README_eng.md

```diff
@@ -0,0 +1,299 @@
+
+And there is a requirement for testing, each version must be covered with tests that are guaranteed to break if someone changes the internal data in any version. 
+
+```rust
+    #[derive(Encode, Decode...)]
+    pub struct TransactionOutput<AccountId> {
+        #[codec(index = 1)]
+        pub(crate) value: Value,
+        #[codec(index = 2)]
+        pub(crate) destination: Destination<AccountId>,
+        #[codec(index = 3)]
+        pub(crate) data: Option<TxData>,
+    }
+
+    #[derive(Encode, Decode...)]
+    pub enum TxData {
+        // TransferToken data to another user. If it is a token, then the token data must also be transferred to the recipient. 
+        #[codec(index = 1)]
+        TransferTokenV1(..),
+        // A new token creation
+        #[codec(index = 2)]
+        TokenIssuanceV1(..),
+        // Burning a token or NFT
+        #[codec(index = 3)]
+        TokenBurnV1(..),
+        // A new NFT creation
+        #[codec(index = 4)]
+        NftMintV1(..),
+        ...
+        // We do not change the data in TransferTokenV1, we just create a new version 
+        #[codec(index = 5)]
+        TransferTokenV2(..),
+    }
+```
+
+## Detailed description of Enum TxData fields
+
+### TransferTokenV1
+---
+
+Transfer funds to another person in a given UTXO. To send MLT we will use `TokenID :: MLT`, which is actually zero. If the `token_id` is equal to the ID of the MLS-01 token, then the amount of token is transferred to the recipient. The commission is taken only in MLT. If the `token_id` is equal to the id of any NFT, then the data of this NFT should be transferred to the recipient without changing the creator field. The UTXO model itself allows you to determine the owner of the NFT. 
+
+```rust
+    pub enum TxData {
+        TransferTokenV1{
+            token_id: TokenID,
+            amount: Value,
+        }
+        ...
+    }
+```
+
+### TokenIssuanceV1
+---
+
+When issuing a new token, we specify the data for creating a new token in the transaction input, where the `token_id` is: 
+
+```rust
+    let token_id = hash(&(inputs[0].hash, inputs[0].index));
+```
+
+`token_ticker` - might be not unique. Should be limited to 10 chars. In fact, it's an ASCII string.
+
+```rust
+    pub enum TxData {
+        TokenIssuanceV1{
+            token_id: TokenID,
+            token_ticker: Vec<u8>,
+            amount_to_issue: Value,
+            // Should be not more than 18 numbers
+            number_of_decimals: u8,
+            metadata_URI: Vec<u8>,
+        }
+        ...
+    }
+```
+See the `metada_URI` format below. 
+
+### TokenBurnV1
+---
+
+Burning a token, as an input is used by UTXO containing tokens, and as an output of BurnV1, in this case, you can burn any number of tokens. After this operation, you can use UTXO for the remaining amount of tokens. 
```

Comment:
I think I missed this. But we won't allow the user to not specify tokens. So, if a user has 1000 TKN in the input, and wants to burn 100, he should make two outputs, one with 100 TKN with burn op, and one with 900 TKN as transfer.